app/models/calculator.rb

```diff
@@ -11,4 +11,8 @@ class Calculator < ApplicationRecord
                              message: 'Only letters and numbers allowed' }
   validates :name, length: { minimum: 2 }
   validates :name, uniqueness: true
+
+  scope :by_name_and_slug, -> (params) { where("name ILIKE ? OR slug ILIKE ?", 
```

Comment:
you use `OR` condition, so rename method to `by_name_or_slug`

Reply:
this approach won't work as expected. 
if `params` is `nil` or empty string it should return all records, without filtering

Reply:
rename, please, `params` to something like `name` or `search`

Reply:
Renamed scope and params in last commit. If `params` is `nil` or empty string it returns all records.